csharp/src/Apache.Arrow.Flight.Sql/PreparedStatement.cs

```diff
@@ -0,0 +1,383 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Apache.Arrow.Flight.Sql.Client;
+using Arrow.Flight.Protocol.Sql;
+using Google.Protobuf;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight.Sql;
+
+public class PreparedStatement : IDisposable
+{
+    private readonly FlightSqlClient _client;
+    private readonly string _handle;
+    private Schema _datasetSchema;
+    private Schema _parameterSchema;
+    private RecordBatch? _recordsBatch;
+    private bool _isClosed;
+    public bool IsClosed => _isClosed;
+    public string Handle => _handle;
+    public RecordBatch? ParametersBatch => _recordsBatch;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PreparedStatement"/> class.
+    /// </summary>
+    /// <param name="client">The Flight SQL client used for executing SQL operations.</param>
+    /// <param name="handle">The handle representing the prepared statement.</param>
+    /// <param name="datasetSchema">The schema of the result dataset.</param>
+    /// <param name="parameterSchema">The schema of the parameters for this prepared statement.</param>
+    public PreparedStatement(FlightSqlClient client, string handle, Schema datasetSchema, Schema parameterSchema)
+    {
+        _client = client ?? throw new ArgumentNullException(nameof(client));
+        _handle = handle ?? throw new ArgumentNullException(nameof(handle));
+        _datasetSchema = datasetSchema ?? throw new ArgumentNullException(nameof(datasetSchema));
+        _parameterSchema = parameterSchema ?? throw new ArgumentNullException(nameof(parameterSchema));
+        _isClosed = false;
+    }
+
+    /// <summary>
+    /// Retrieves the schema associated with the prepared statement asynchronously.
+    /// </summary>
+    /// <param name="options">The options used to configure the Flight call.</param>
+    /// <returns>A task representing the asynchronous operation, which returns the schema of the result set.</returns>
+    /// <exception cref="InvalidOperationException">Thrown when the schema is empty or invalid.</exception>
+    public async Task<Schema> GetSchemaAsync(FlightCallOptions? options = default)
+    {
+        EnsureStatementIsNotClosed();
+
+        try
+        {
+            var command = new CommandPreparedStatementQuery
+            {
+                PreparedStatementHandle = ByteString.CopyFrom(_handle, Encoding.UTF8)
+            };
+            var descriptor = FlightDescriptor.CreateCommandDescriptor(command.PackAndSerialize());
+            var schema = await _client.GetSchemaAsync(descriptor, options).ConfigureAwait(false);
+            if (schema == null || !schema.FieldsList.Any())
+            {
+                throw new InvalidOperationException("Schema is empty or invalid.");
+            }
+            return schema;
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to retrieve the schema for the prepared statement", ex);
+        }
+    }
+
+
```

Comment:
nit: extra blank line

Reply:
don't the extra line, but removed one above.